### ISSUE TITLE
[Bug] Stack Corruption Detection Fails on Valid Cases

### ISSUE THREAD

**Issue by jp2masa** [CosmosOS/IL2CPU#8]
The stack corruption detection fails when there are complex conditions, because there are branches which implicitly assume that some IL instructions will only be executed on certain conditions (specific stack differences).

A sample, in C#: [SharpLab](https://sharplab.io/#v2:C4LglgNgNAJiDUAfAAgBgATIIwG4CwAUIcgMyYBM6AwoQN6HqOZnYBsmALJlgBwAUASgZN6BJuPQAjAPbSI6AGYQAhgHMs6ALzpgAJwCuAU3xiJjGXMUrVlbQuUQAzseFmL8pWrLa9Rk2cZXCTAFdD5PdXQAMiiwiMpERCsvASFTM1EAiWwATj4AIgAVA0N0AGNpADsYMGAwKvyBfwCAXyDxQydDdpEes1yCgDEHZ3KqmrqGpr70NvTZwhagA===)

<details>
<summary>Analyzing the C# sample (IL, using 32-bit sizes)</summary>

```IL
... truncated ... -> stack difference = base
ΔSD -> Δ Stack Difference
[ΔSD]
[ +4]IL_0007: ldloc.0
[ -4]IL_0008: brfalse.s IL_000f          // stack difference = base
[ +4]IL_000a: ldloc.1
[ +4]IL_000b: ldloc.2
[ -4]IL_000c: or
[  0]IL_000d: br.s IL_0010               // stack difference = base + 4
[ +4]IL_000f: ldc.i4.0                   // came from IL_0008 -> stack difference before op = base
[ -4]IL_0010: stloc.3                    // came from IL_000d -> stack difference before op = base + 4
[ +4]IL_0011: ldloc.3
[ -4]IL_0012: brfalse.s IL_0023          // stack difference = base
                                         // calculated stack difference: base + 4
                                         // stack corruption on IL_0014
[  0]IL_0014: nop
[ +8]IL_0015: ldstr "True condition"
[ -8]IL_001a: call void [mscorlib]System.Console::WriteLine(string)
[  0]IL_001f: nop
[  0]IL_0020: nop
[  0]IL_0021: br.s IL_0030               // stack difference -> base
[  0]IL_0023: nop                        // came from IL_0012 -> stack difference before op = base
[ +8]IL_0024: ldstr "False condition"
[ -8]IL_0029: call void [mscorlib]System.Console::WriteLine(string)
[  0]IL_002e: nop
[  0]IL_002f: nop
[  0]IL_0030: ret
```
</details>

**Comment by wessupermare** [comment]
Hello! I'm having trouble reproducing this bug. I ran the supplied code through the current dev build of IL2CPU and it built and ran with no errors or even warnings. Is it possible that it's been fixed in the past year? If you are still able to reproduce, please include a stack trace.

**Comment by quajak** [comment]
Is fixed by #94
